app/controllers/dashboard/form/publish_controller.rb

```diff
@@ -25,7 +25,8 @@ def update
         # save again--this time using the published validations. That way the
         # appropriate error messages will appear on the form when it's re-rendered
         if publish?
-          save_resource(index: false)
+          @resource.indexing_source = Proc.new { nil }
+          @resource.save
```

Comment:
What is the significance of this change? Looks like it now skips the `@resource.update_doi` stuff from `save_resource` now.

Reply:
This is the bug you were noticing earlier. We save the resource twice. The first time is to catch validation errors that occur during publishing, and to make sure that changes are saved to the database, even if it can't be published. There's a comment above this method that explains that, it's just not showing up in the diff.

I'm not sure why this wasn't a problem earlier, but the `Proc.new { nil }` is so that the work version is not indexed during this initial save-check.